tempesta_fw/http.c

```diff
@@ -272,6 +278,12 @@ tfw_http_send_resp(TfwHttpReq *req, TfwStr *msg, const TfwStr *date)
 	tfw_http_resp_fwd(req, (TfwHttpResp *)hmresp);
 
 	return 0;
+mem_err:
+	TFW_DBG2("%s: Memory allocation error: conn=[%p]\n",
+		 __func__, req->conn);
+	ss_close_sync(req->conn->sk, true);
+	tfw_http_conn_msg_free((TfwHttpMsg *)req);
+	return -ENOMEM;
```

Comment:
Also, exactly the same situation is processed here and in zap_error(), it should be moved into separate function to avoid code duplication.

Reply:
I believe a short comment would help to understand the consequences of closing a client connection here.

As soon there's an error sending a response, the pairing of requests and responses gets broken, and the connection must be closed to avoid wrong pairings with subsequent responses.

At the same time, as soon as a prepared response is not linked with a paired request, sending to that client stops starting with that request, because that creates a "hole" in the chain of requests -- a request without a response. Subsequent responses cannot be sent to the client until that hole is closed, which at this point will never happen.

Reply:
Comment added.